### PR TITLE
[fix] Missing WWW-Authenticate header in gatekeeper_access_denied_response()

### LINKED ISSUE #81: The WWW-Authenticate header is missing from HTTP 401 responses from gatekeeper
The bug slipped through because we don't have any tests for methods from [utils.py](https://github.com/golemfactory/concent/blob/master/concent_api/gatekeeper/utils.py). Add some. Also add assertions the check the presence of the header in existing gatekeeper tests.

### PR DESCRIPTION
**Pull request by bartoszbetka**
Resolves #81

### CODE REVIEW

File: concent_api/gatekeeper/tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+class GatekeeperAccessDeniedResponseTest(TestCase):
+
+    def setUp(self):
+        self.message     = "Missing 'Authorization' header."
+        self.path        = 'blender/benchmark/test_task/scene-Helicopter-27-cycles.blend'
+        self.client_key  = '85cZzVjahnRpUBwm0zlNnqTdYom1LF1P1WNShLg17cmhN2UssnPrCjHKTi5susO3wrr/q07eswumbL82b4HgOw=='
+
+    def test_response_should_return_appropriate_body_and_header(self):
+        response =  gatekeeper_access_denied_response(
```
Comment (rwrzesien) [inline concent_api/gatekeeper/tests/test_utils.py]: Please remove extra space after ` = `.

File: concent_api/gatekeeper/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                self.message,
```
Comment (rwrzesien) [inline concent_api/gatekeeper/tests/test_utils.py]: Please move indent 4 spaces to the left.

File: concent_api/gatekeeper/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            self.path,
+            None,
+            self.client_key
+        )
+        response_body = json.loads(response.content)
+
+        self.assertIsInstance(response, JsonResponse)
+        self.assertEqual(response.status_code, 401)
+        self.assertTrue(response.has_header("WWW-Authenticate"))
+        self.assertEqual("application/json", response["Content-Type"])
```
Comment (cameel) [inline concent_api/gatekeeper/tests/test_utils.py]: By "use `assertIn`" I meant a check that ensures that `"Content-Type"` key is present in `response`. Please add it.

File: concent_api/gatekeeper/tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+from django.test                    import TestCase
+
+from gatekeeper.utils               import gatekeeper_access_denied_response
+
+class GatekeeperAccessDeniedResponseTest(TestCase):
+
+    def setUp(self):
+        self.message     = "Missing 'Authorization' header."
+        self.path        = 'blender/benchmark/test_task/scene-Helicopter-27-cycles.blend'
+        self.client_key  = '85cZzVjahnRpUBwm0zlNnqTdYom1LF1P1WNShLg17cmhN2UssnPrCjHKTi5susO3wrr/q07eswumbL82b4HgOw=='
```
Comment (cameel) [inline concent_api/gatekeeper/tests/test_utils.py]: Don't hard-code keys if not necessary. Use our `generate_key_pair()` helper.

File: concent_api/gatekeeper/tests/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    def test_response_should_return_appropriate_body_and_header(self):
```
Comment (cameel) [inline concent_api/gatekeeper/tests/test_utils.py]: You're testing a function called `gatekeeper_access_denied_response` so the test should be called `test_gatekeeper_access_denied_response_should_return_appropriate_body_and_headers()` to follow our convention.

File: concent_api/gatekeeper/tests/test_utils.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import json
+
+from django.http                    import JsonResponse
+from django.test                    import TestCase
+
+from gatekeeper.utils               import gatekeeper_access_denied_response
+
+class GatekeeperAccessDeniedResponseTest(TestCase):
+
+    def setUp(self):
```
Comment (cameel) [inline concent_api/gatekeeper/tests/test_utils.py]: `full-check.sh` reports a warning for this line:

```
./concent_api/gatekeeper/tests/test_utils.py:8:1: E302 expected 2 blank lines, found 1
```

Before submitting a pull request please make sure that your code passes all tests and does not generate any new warnings.

**Pull request merged**